api/src/main/java/uk/gov/dhsc/htbhf/claimant/reporting/MIReporter.java

```diff
@@ -30,23 +31,35 @@
     private final GoogleAnalyticsClient googleAnalyticsClient;
 
     public void reportClaim(ReportClaimMessageContext context) {
-        updateClaimWithPostcodeDataIfNecessary(context);
+        Claim claim = context.getClaim();
+        if (postcodeDataIsNotSet(claim) || addressHasBeenUpdated(context)) {
+            retrievePostcodeDataAndSaveToClaim(claim);
+        }
         Map<String, String> reportProperties = reportClaimPropertiesFactory.createReportPropertiesForClaimEvent(context);
         googleAnalyticsClient.reportEvent(reportProperties);
     }
 
     public void reportPayment(ReportPaymentMessageContext context) {
-        updateClaimWithPostcodeDataIfNecessary(context);
+        Claim claim = context.getClaim();
+        if (claim.getPostcodeData() == null) {
```

Comment:
You can use `postcodeDataIsNotSet(claim)` here too.

Reply:
Done.